File: indexer/extract/generic/testdata/kotlin/sample.kt
package com.example.store

import kotlin.collections.List
import com.example.base.*
import com.example.util.Helper as H

/** A user repository. */
@Service
class UserRepo(private val db: Database) : BaseRepo<User>(), Repo, AutoCloseable {
    private val cache = mutableMapOf<String, User>()
    internal var count: Int = 0

    override fun find(id: String): User? {
        H.check(id)
        val u = User(id)
        return cache[id] ?: db.load(id)
    }

    @Deprecated("old")
    suspend fun legacy() {}

    companion object {
        const val LIMIT = 10
        fun create(): UserRepo = UserRepo(Database())
    }
}

interface Repo {
    fun find(id: String): User?
}

data class User(val id: String)

enum class Mode { READ, WRITE }

object Registry {
    fun register(r: Repo) {}
}

typealias Id = String

fun topLevel(x: Int): Int = helper(x)

private fun helper(x: Int) = x + 1

File: indexer/extract/generic/testdata/scala/sample.scala
package com.example.store

import scala.collection.mutable
import com.example.base.{BaseRepo, Repo => R}
import com.example.util._

/** A user repository. */
class UserRepo(db: Database) extends BaseRepo[User] with Repo with AutoCloseable {
  private val cache = mutable.Map[String, User]()
  protected var count: Int = 0

  override def find(id: String): Option[User] = {
    Helper.check(id)
    val u = new User(id)
    cache.get(id)
  }

  @deprecated("old", "1.0")
  def legacy(): Unit = ()
}

trait Repo {
  def find(id: String): Option[User]
}

case class User(id: String)

object UserRepo {
  val Limit = 10
  def apply(db: Database): UserRepo = new UserRepo(db)
}

type Id = String

def topLevel(): Unit = println("x")
